3-reveal/migrations/5-IRS/3-Namibia-2019/deploy/redo_namibia_irs_structures.psql

```diff
@@ -0,0 +1,122 @@
+        subq.refusalReasonMop AS refusalReasonMop,
+        subq.business_status AS business_status,
+        array_agg(subq.unsprayedReasonFirst) FILTER (WHERE subq.unsprayedReasonFirst != '') OVER (PARTITION BY unsprayedReasonFirst) AS unsprayedReasonFirst_values,
+        array_agg(subq.refusalReasonFirst) FILTER (WHERE subq.refusalReasonFirst != '') OVER (PARTITION BY refusalReasonFirst) AS refusalReasonFirst_values,
+        array_agg(subq.unsprayedReasonMop) FILTER (WHERE subq.unsprayedReasonMop != '') OVER (PARTITION BY unsprayedReasonMop) AS unsprayedReasonMop_values,
+        array_agg(subq.refusalReasonMop) FILTER (WHERE subq.refusalReasonMop != '') OVER (PARTITION BY refusalReasonMop) AS refusalReasonMop_values
+    FROM
+    (
+        SELECT
+            DISTINCT ON (tasks.identifier)
+            events.id AS event_id,
+            tasks.identifier AS task_id,
+            events.event_date AS event_date,
+            tasks.plan_identifier AS plan_id,
+            -- get fields from JSON --
+            COALESCE (events.form_data->'nSprayableTotal'->>0, '0')::int AS nSprayableTotal,
+            COALESCE (events.form_data->'nSprayableZinc'->>0, '0')::int AS nSprayableZinc,
+            COALESCE (events.form_data->'nSprayableTrad'->>0, '0')::int AS nSprayableTrad,
+            COALESCE (events.form_data->'nSprayableMod'->>0, '0')::int AS nSprayableMod,
+            COALESCE (events.form_data->'nSprayableCanopy'->>0, '0')::int AS nSprayableCanopy,
+            COALESCE (events.form_data->'nUnsprayable'->>0, '0')::int AS nUnsprayable,
+            COALESCE (events.form_data->'nUnsprayedTotalFirst'->>0, '0')::DECIMAL AS nUnsprayedTotalFirst,
+            COALESCE (events.form_data->'nUnsprayedTotalMop'->>0, '0')::DECIMAL AS nUnsprayedTotalMop,
+            COALESCE (events.form_data->'nSprayedTotalFirst'->>0, '0')::int AS nSprayedTotalFirst,
+            COALESCE (events.form_data->'nSprayedTotalMop'->>0, '0')::int AS nSprayedTotalMop,
+            COALESCE (events.form_data->'householdAccessible'->>0, '')::text AS householdAccessible,
+            COALESCE (events.form_data->'unsprayedReasonFirst'->>0, '')::text AS unsprayedReasonFirst,
+            COALESCE (events.form_data->'refusalReasonFirst'->>0, '')::text AS refusalReasonFirst,
+            COALESCE (events.form_data->'unsprayedReasonMop'->>0, '')::text AS unsprayedReasonMop,
+            COALESCE (events.form_data->'refusalReasonMop'->>0, '')::text AS refusalReasonMop,
+            COALESCE (events.form_data->'business_status'->>0, 'Not Visited')::text AS business_status
+            -- end get fields from JSON --
+        FROM tasks
+        LEFT JOIN events
+            ON tasks.identifier = events.task_id
+            AND events.entity_type = 'Structure'
+            AND events.event_type = 'Spray'
+        WHERE locations.id = tasks.task_for
+        AND tasks.status != 'Cancelled'
+        ORDER BY task_id, events.form_data->>'end' DESC
```

Comment:
It's hard to tell, but this single line is really the logical change, right?

Reply:
Yes it is.